compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/ExpressionCodegen.kt

```diff
@@ -202,6 +208,21 @@ class ExpressionCodegen(
         )
     }
 
+    private fun writeReceiverParameterInLocalVariableTable(param: IrValueParameter, startLabel: Label, endLabel: Label) {
```

Comment:
As name is the only thing that changes, I'd move the code back to `writeValueParameterInLocalVariableTable()` and added a `when` (or moved only the name calculation to the separate function).